docs/source/tutorials/building_blocks.rst

```diff
@@ -294,6 +294,75 @@ Use the :code:`run_time` argument to control the duration.
+Manim has a feature that allows you to customize this behaviour by defining your own custom animation.
+
+You can start by creating your own Count class that extends :class:`~.Animation` class.
+The class can have a constructor with three arguments, :class:`~.DecimalNumber` object, start and end.
+The constructor will pass the :class:`~.DecimalNumber` object to super constructor and will set up start and end.
+
+The only thing that you need to do is to define the logic of output at every step of the animation.
+Manim provides you with the alpha value in the :meth:`~.Animation.interpolate_mobject` method based on frame rate of video, rate function and run time of animation played.
+The alpha parameter holds a value between 0 and 1 representing the step of currently playing animation.
+For example, 0 means beginning of the animation, 0.5 means half way through the animation, 1 means end of animation and so on.
+
+In case of the Count animation, you just have to figure out a way to determine the number to display at the given alpha value and then set that value in the :meth:`~.Animation.interpolate_mobject` method of Count Animation.
+Suppose you are starting at 50 and incrementing till 100 at the end of the animation.
+If alpha is 0, you want value to be 50.
+If alpha is 0.5, you want value to be 75.
+If alpha is 1, you want value to be 100.
+So the logic of calcuating the number to display at each step will be - 50 + alpha * (100 - 50).
+Set the calculated value for decimal number and you are done.
+
+Once, you have defined Count Animation, you can play it in your :class:`~.Scene` for any duration you want for any DecimalNumber with any rate function.
+
+.. manim:: CountingScene
+    :quality: medium
+    :ref_classes: Animation DecimalNumber
+    :ref_methods: Animation.interpolate_mobject Scene.play
+
+    class Count(Animation):
+        def __init__(self, number: DecimalNumber, start: float, end: float, ** kwargs) -> None:
+            # Pass num as mobject of animation
+            super().__init__(number,  **kwargs)
+            # Set start and end
+            self.start = start
+            self.end = end
+
+        def interpolate_mobject(self, alpha: float) -> None:
+            # Set value of DecimalNumber according to alpha
+            value = self.start + (alpha * (self.end - self.start))
+            self.mobject.set_value(value)
+
+
+    class CountingScene(Scene):
+        def construct(self):
+            # Create Decimal Number and add it to scene
+            number = DecimalNumber().set_color(WHITE).scale(5)
+            number.add_updater(lambda number: number.move_to(ORIGIN))
+            self.add(number)
+
+            # Wait for 1 second
+            self.wait(1)
+
+            # Play Count Animation to count from 0 to 100 in 4 seconds
+            self.play(Count(number, 0, 100), run_time=4, rate_func=linear)
```

Comment:
Is there any reason to specify `linear`?

Reply:
I just wanted to make it fit with my explanation since my explanation mentions that alpha is 0 at the start, 0.5 in mid, and 1 at the end. For other rate functions, that's not the case.